src/gui/generalsettings.cpp

```diff
@@ -282,10 +283,15 @@ void GeneralSettings::saveMiscSettings()
         // for now, we use the locale names as labels in the dropdown
         // therefore, we can store them directly in the config file
         // in future versions, we will likely display nice names instead of locales to improve the user experience
-        const auto pickedLanguageName = _ui->languageDropdown->itemText(pickedLanguageIndex);
-        const auto pickedLanguageLocale = localesToLanguageNamesMap.key(pickedLanguageName);
 
-        cfgFile.setUiLanguage(pickedLanguageLocale);
+        QRegularExpression regex(".+\\s+\\(([^)]+)\\)");
+        Q_ASSERT(regex.isValid());
+
+        auto match = regex.match(_ui->languageDropdown->itemText(pickedLanguageIndex));
```

Comment:
Comboboxes also provide https://doc.qt.io/qt-5/qcombobox.html#itemData which could hold the file name or raw language etc, might be more elegant than the regex.

Reply:
Good hint. I was looking for something like this before, actually. Must have missed it.